python/xrprimer/utils/visualization_utils.py

```diff
@@ -72,3 +89,108 @@ def fix_arr_shape(
                          f' getting {array.shape}.')
             raise ValueError
     return array
+
+
+def check_data_len(mframe_point_data: Union[np.ndarray, None],
+                   mframe_line_data: Union[np.ndarray, None],
+                   background_len: int, logger: logging.Logger) -> int:
```

Comment:
What about a more general one? The argument for this function is a list, where each element is a np.ndarray or int.

Reply:
The input parameter checking is currently divided into two steps.
In the first step, based on all the background-related parameters, it is checked whether the number of background parameters is correct and whether it reflects the number of background frames.
In the second step, based on the number of background frames (which can be empty), point frames, and line frames, it is checked whether the number of point and line parameters is correct and whether the frame numbers are all equal.

If the two steps are combined, parameters other than width and height can be defined using the same format, but the function will become longer and its versatility will decrease. If not combined, the three parameters in the second step are not equal in status.

In the sencond step, there are a few checks:
1. no less than 1 is provided between `mframe_point_data` and `mframe_line_data`
2. any provided length must be equal.

Reply:
According to discussion offline, there will be 2 argument number check functions and one argument length check function.

Reply:
Fixed in [881d05c](https://github.com/openxrlab/xrprimer/pull/39/commits/881d05c6d5885e41d6b97ff1bd4567b104c387fb).